### PR TITLE
Use babel to transpile typescript, remove global polyfill and add modern bundle

### PR DESCRIPTION
**Pull request by m1aw**
## Summary

This ticket has 3 main objectives:
- Remove the pollution happening with globally scoped polyfills
- Atomically handle new polyfills
- Provide an unpolyfilled / "modern" version of the lib to reduce the transferred size

To achieve this we start using core-js-pure via babel-runtime, this explains why babel was introduced. 
For fast babel transpilation we started using babel-typescript-preset instead of doing 2 stages (tsc then babel).
For the "modern" build we use babel and babel-preset env with `target:"module"` and `bugfixes:true`.
Also the umd build now only runs when building the production bundle and not in dev to speed up compilation.

## Tested scenarios

- Unit test passed successfully 
- There was no increase in failed e2e tests
- The resulting bundles where tests on chrome for global scoped pollution (checked if Function.toString() was being polyfilled for example)
- Tested normal dropin flow on IE11

**Fixed issue**:  

Might help #337

### CODE REVIEW

File: packages/lib/config/rollup.config.js [hunk 1]
```diff
@@ -32,9 +33,30 @@ const watchConfig = {
...
 
-async function getPlugins({ compress, analyze, version, useTypescript = true }) {
+const extensions = ['.js', '.jsx', '.ts', '.tsx'];
+
+const polyfillPlugin = [
+    '@babel/plugin-transform-runtime',
+    {
+        corejs: 3,
+        absoluteRuntime: true
+        //useESModules: true
```
Comment (marcperez) [inline packages/lib/config/rollup.config.js:43]: Is this necessary? Can it go?

File: packages/playground/config/polyfills.js [hunk 2]
```diff
@@ -19,10 +19,3 @@ import 'whatwg-fetch';
         });
     });
 })([Element.prototype, CharacterData.prototype, DocumentType.prototype]);
```
Comment (marcperez) [inline packages/playground/config/polyfills.js:21]: Will the new polyfills remove the need for this `ChildNode.remove()` one?

Comment (marcperez) [reply packages/playground/config/polyfills.js:21]: (same for `fetch`)

Comment (ribeiroguilherme) [reply packages/playground/config/polyfills.js:21]: If `fetch` is polyfiled by `core-js-pure`, we might be able to remove the dependency `whatwg-fetch`

Comment (ribeiroguilherme) [reply packages/playground/config/polyfills.js:21]: If `fetch` is polyfiled by `core-js-pure`, we might be able to remove the dependency `whatwg-fetch`

Comment (m1aw) [reply packages/playground/config/polyfills.js:21]: I just left this for the playground because there is still code we are not transpilling in there.

Comment (ribeiroguilherme) [reply packages/playground/config/polyfills.js:21]: Sorry, I wasn't clear enough. I didn't mean in the playground; we use `whatg-fetch` as polyfill in our `lib` , on `src/core/Services/fetch.ts` . Can we ditch that polyfill too?

Comment (m1aw) [reply packages/playground/config/polyfills.js:21]: we cannot remove this, using corejs there where errors on IE11

**Review by marcperez — APPROVED** [review]

**Comment by lynn1286** [comment]
image

我在使用nextjs构建adyen ，但是 webpack chunks 异常庞大，是我的使用方式有误吗？
下面是我的package.json

```
"@adyen/adyen-web": "^5.15.0",
 "next": "12.1.6",
```

下面是我的代码：
```
import type { ChangeEventHandler } from 'react'
import React, { useEffect, useRef, useState } from 'react'
import { useRouter } from 'next/router'
import classnames from 'classnames'
import useSWR from 'swr'
import adyenCheckout from '@adyen/adyen-web'
import type Dropin from '@adyen/adyen-web/dist/types/components/Dropin'
import type { PaymentResponse } from '@adyen/adyen-web/dist/types/types'
import { paymentMethodsApi } from 'https/adyen-request'
import MessageAlert from '@/components/message/index'
import { usePaymentState } from '@/hooks/index'
import type { SessionsResponse } from '@/types/adyen'
import type { BillingAddress, ResultCode } from '@/hooks/use-payment-state'
import { PaymentStateAction, PaymentModelState } from '@/hooks/use-payment-state'
import '@adyen/adyen-web/dist/adyen.css'
```

**Pull request merged**